src/main/java/uk/gov/digital/ho/hocs/audit/export/CustomExportService.java

```diff
@@ -65,21 +65,29 @@ public void customExport(HttpServletResponse response, String code, boolean conv
             if (convertHeader) {
                 substitutedHeaders = headerConverter.substitute(headers);
             }
-            List<Object[]> dataList = customExportDataConverter.convertData(auditRepository.getResultsFromView(exportViewDto.getCode()), exportViewDto.getFields());
-
-            try (CSVPrinter printer = new CSVPrinter(outputWriter, CSVFormat.DEFAULT.withHeader(substitutedHeaders.toArray(new String[substitutedHeaders.size()])))) {
-
-                dataList.forEach((dataRow) -> {
-                    try {
-                        printer.printRecord(dataRow);
-                        outputWriter.flush();
-                    } catch (Exception e) {
-                        log.error("Unable to parse record for custom report, reason: {}, event: {}", e.getMessage(), value(LogEvent.EVENT, CSV_EXPORT_FAILURE));
-                    }
-                });
-                log.info("Export Custom Report '{}' to CSV Complete, event {}", exportViewDto.getCode(), value(EVENT, CSV_EXPORT_COMPETE));
-            }
 
+            Stream<Object[]> dataList = auditRepository
+                    .getResultsFromView(exportViewDto.getCode())
+                    .map(data ->
+                        customExportDataConverter.convertData(data, exportViewDto.getFields())
+                    );
+            createCSV(outputWriter, substitutedHeaders, dataList);
```

Comment:
This process increases the throughput and we can move forward with approving it for testing in dev.
As a next step it would be optimal to include the createCSV output processing in the map above to ensure continuous streaming at a record granularity.

Reply:
I agree with this - this work will commence immediately.

Reply:
This work has been completed - still need to look at how we can seperate the reading and writing efficiently.